src/dataSources/cloudFirestore/community.js

```diff
@@ -119,24 +125,73 @@ const community = dbInstance => {
         id: d.id,
         ...d.data(),
       };
+      result = communityDateForge(result);
     } else if (size > 1) {
       throw new Error(`Multiple Community slugs found for slug ${slimslug}`);
     }
 
     return result;
   }
 
-  async function isSlugTaken(slug) {
-    dlog('isSlugTaken? %s', slug);
-    const { size } = await communityCol
-      .where('slug', '==', slug)
-      .select()
-      .get();
-
-    return size > 0;
+  function isSlugTaken(slug) {
+    dlog('isSlugTaken called %s', slug);
+    return slugStore(dbInstance).isSlugTaken(slug);
   }
 
   async function create({ newCommunity, user }) {
+    dlog('create new community with slug %s', newCommunity.slug);
+    const cleanCommunity = scrubCommunity({
+      community: newCommunity,
+      isNew: true,
+      user,
+    });
+    const newSlug = cleanCommunity.slug;
+    const slugInUse = await isSlugTaken(newSlug);
+    if (slugInUse)
+      throw new Error(
+        'Slug in use, it cannot be used to create a new community',
+      );
+
+    const communityDocRef = communityCol.doc(); // creates a random id
+    dlog('new community id %s');
+    const slugDocRef = slugStore(dbInstance).getSlugDocRef(newSlug);
```

Comment:
same here with batching these calls

Reply:
getSlugRef doesn't return a Promise. It is a reference created by the sdk. This reference is needed for the batch write to Firestore